### PR TITLE
DDF-1886 Merge ebrim transformer updates from ddf-registry to ddf

### PR DESCRIPTION
**Pull request by mcalcote**
#### What does this PR do?
This PR merges in changes that were made to the ebrim transformer during development on the ddf-registry branch. 

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@harrison-tarr @spearskw 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@pklinef
@shaundmorris 

#### How should this be tested?

Please review the code, build and the run tests.

#### What are the relevant tickets?
DDF-2109 - Registry package converter uses Registry Package Utils 

#### Screenshots (if appropriate)
Backend Functionality

#### Checklist:
- [ ] Documentation Updated
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

**Comment by shaundmorris** [comment]
👍  pending hero and if there are necessary changes from DDF-2106

File: catalog/spatial/registry/registry-ebrim-transformer/src/main/java/org/codice/ddf/registry/converter/RegistryPackageConverter.java [hunk 1]
```diff
@@ -414,12 +405,10 @@ private static void parseTopLevel(RegistryObjectType registryObject, MetacardImp
...
-            for (ExternalIdentifierType extId : extIds) {
-                if (extId.getId()
-                        .equals(RegistryConstants.REGISTRY_MCARD_LOCAL_ID)) {
-                    metacard.setId(extId.getValue());
-                }
-            }
+            extIds.stream()
+                    .filter(extId -> extId.getId()
+                            .equals(RegistryConstants.REGISTRY_MCARD_LOCAL_ID))
+                    .forEach(extId -> metacard.setId(extId.getValue()));
```
Comment (harrison-tarr) [inline catalog/spatial/registry/registry-ebrim-transformer/src/main/java/org/codice/ddf/registry/converter/RegistryPackageConverter.java]: how many extIds do you expect to match? It looks like it's just one (or if multiple match, you overwrite the metacard ID multiple times) I think you could use a stream().findFirst() call here

Comment (mcalcote) [inline catalog/spatial/registry/registry-ebrim-transformer/src/main/java/org/codice/ddf/registry/converter/RegistryPackageConverter.java]: With this and the comment on the other PR, I decided to look into stream efficiency and it actually looks a bit worse than normal loops. The trade off is (potentially) cleaner code which may be safer.   :|

Will look into findFirst for here though.

Comment (spearskw) [inline catalog/spatial/registry/registry-ebrim-transformer/src/main/java/org/codice/ddf/registry/converter/RegistryPackageConverter.java]: You're a mad man!

I think for-loops are fine, and perfectly readable (sometimes moreso for me). Also, if you only want 1 match and it doesn't matter which, you can add a `break;`.

**Comment by ryeats** [comment]
hero'd 👍

**Pull request merged**